internal/checker/emitresolver.go

```diff
@@ -1,21 +1,586 @@
+						continue // no alias to add, already exported
+					}
+					if !r.isDeclarationVisible(variableStatement.Parent) {
+						return nil // not visible
+					}
+					addVisibleAlias(declaration, variableStatement)
+					continue
+				}
+			}
+
+			// Declaration is not visible
+			return nil
+		}
+	}
+
+	return &printer.SymbolAccessibilityResult{
+		Accessibility:        printer.SymbolAccessibilityAccessible,
+		AliasesToMakeVisible: slices.Collect(maps.Values(aliasesToMakeVisibleSet)),
+	}
+}
+
+func (r *emitResolver) IsImplementationOfOverload(node *ast.SignatureDeclaration) bool {
+	// node = r.emitContext.ParseNode(node)
+	if !ast.IsParseTreeNode(node) {
+		return false
+	}
+	if ast.NodeIsPresent(node.Body()) {
+		if ast.IsGetAccessorDeclaration(node) || ast.IsSetAccessorDeclaration(node) {
+			return false // Get or set accessors can never be overload implementations, but can have up to 2 signatures
+		}
+		r.checkerMu.Lock()
+		defer r.checkerMu.Unlock()
+		symbol := r.checker.getSymbolOfDeclaration(node)
+		signaturesOfSymbol := r.checker.getSignaturesOfSymbol(symbol)
+		// If this function body corresponds to function with multiple signature, it is implementation of overload
+		// e.g.: function foo(a: string): string;
+		//       function foo(a: number): number;
+		//       function foo(a: any) { // This is implementation of the overloads
+		//           return a;
+		//       }
+		return len(signaturesOfSymbol) > 1 ||
+			// If there is single signature for the symbol, it is overload if that signature isn't coming from the node
+			// e.g.: function foo(a: string): string;
+			//       function foo(a: any) { // This is implementation of the overloads
+			//           return a;
+			//       }
+			(len(signaturesOfSymbol) == 1 && signaturesOfSymbol[0].declaration != node)
+	}
+	return false
+}
+
+func (r *emitResolver) IsImportRequiredByAugmentation(decl *ast.ImportDeclaration) bool {
+	// node = r.emitContext.ParseNode(node)
+	if !ast.IsParseTreeNode(decl.AsNode()) {
+		return false
+	}
+	file := ast.GetSourceFileOfNode(decl.AsNode())
+	if file.Symbol == nil {
+		// script file
+		return false
+	}
+	importTarget := r.GetExternalModuleFileFromDeclaration(decl.AsNode())
+	if importTarget == nil {
+		return false
+	}
+	if importTarget == file {
+		return false
+	}
+	r.checkerMu.Lock()
+	defer r.checkerMu.Unlock()
+	exports := r.checker.getExportsOfModule(file.Symbol)
+	for s := range maps.Values(exports) {
+		merged := r.checker.getMergedSymbol(s)
+		if merged != s {
+			if len(merged.Declarations) > 0 {
+				for _, d := range merged.Declarations {
+					declFile := ast.GetSourceFileOfNode(d)
+					if declFile == importTarget {
+						return true
+					}
+				}
+			}
+		}
+	}
+	return false
+}
+
+func (r *emitResolver) RequiresAddingImplicitUndefined(declaration *ast.Node, symbol *ast.Symbol, enclosingDeclaration *ast.Node) bool {
+	// node = r.emitContext.ParseNode(node)
+	if !ast.IsParseTreeNode(declaration) {
```

Comment:
All of these `IsParseTreenode` calls appear to be dead (at least, uncovered); is this just since we don't have anyone calling us with weird nodes?